aws-logging-cloudwatch/src/main/java/com/amplifyframework/logging/cloudwatch/AWSCloudWatchLoggingPlugin.kt

```diff
@@ -0,0 +1,138 @@
+    private val awsCloudWatchLoggingPluginImplementation = AWSCloudWatchLoggingPluginImplementation(
+        loggingConstraintsResolver,
+        awsCloudWatchLoggingPluginConfig,
+    )
+    private lateinit var cloudWatchLogsClient: CloudWatchLogsClient
+    private val logger = Amplify.Logging.logger(CategoryType.LOGGING, this::class.java.simpleName)
+
+    companion object {
+        private const val CONFIG_FILENAME = "amplifyconfiguration_logging"
+        internal const val SHARED_PREFERENCE_FILENAME = "com.amplify.logging.a3fa4188-0ac5-11ee-be56-0242ac120002"
+    }
+
+    @Deprecated("Deprecated in Java")
+    override fun forNamespace(namespace: String?): Logger {
+        return awsCloudWatchLoggingPluginImplementation.forNamespace(namespace)
+    }
+
+    override fun logger(namespace: String): Logger {
+        return awsCloudWatchLoggingPluginImplementation.logger(namespace)
+    }
+
+    override fun logger(categoryType: CategoryType, namespace: String): Logger {
+        return awsCloudWatchLoggingPluginImplementation.logger(categoryType, namespace)
+    }
+
+    override fun enable() {
+        awsCloudWatchLoggingPluginImplementation.enable()
+    }
+
+    override fun disable() {
+        awsCloudWatchLoggingPluginImplementation.disable()
+    }
+
+    public fun flushLogs(
+        onSuccess: Action,
+        onError: Consumer<AmplifyException>,
+    ) {
+        awsCloudWatchLoggingPluginImplementation.flushLogs(onSuccess, onError)
+    }
+
+    override fun getPluginKey(): String {
+        return "awsCloudWatchLoggingPlugin"
+    }
+
+    override fun configure(pluginConfiguration: JSONObject?, context: Context) {
+        try {
+            val awsLoggingConfig = awsCloudWatchLoggingPluginConfig ?: getConfigFromFile(pluginConfiguration)
+            loggingConstraintsResolver.context = context
+            cloudWatchLogsClient = CloudWatchLogsClient {
+                credentialsProvider = CognitoCredentialsProvider()
+                region = awsLoggingConfig.region
+            }
+            if (awsRemoteLoggingConstraintProvider == null) {
+                awsLoggingConfig.defaultRemoteConfiguration?.let {
+                    loggingConstraintsResolver.setRemoteConfigProvider(
+                        DefaultRemoteLoggingConstraintProvider(
+                            URL(it.endpoint),
+                            awsLoggingConfig.region,
+                            it.refreshIntervalInSeconds,
+                        ),
+                    )
+                }
+            }
+            val cloudWatchLogManager =
+                CloudWatchLogManager(context, awsLoggingConfig, cloudWatchLogsClient, loggingConstraintsResolver)
+            awsCloudWatchLoggingPluginImplementation.cloudWatchLogManager = cloudWatchLogManager
+            awsCloudWatchLoggingPluginImplementation.configure(awsLoggingConfig)
+        } catch (exception: Exception) {
+            logger.error("failed to configure plugin", exception)
```

Comment:
This should throw an AmplifyException: https://github.com/aws-amplify/amplify-android/blob/main/aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt#L146